app/src/main/java/com/example/atheneum/activities/ViewEditBookPhotos.java

```diff
@@ -0,0 +1,157 @@
+            Log.w(TAG, "ownerUserID is not valid! This should never happen!");
+            return;
+        } else if (!FirebaseAuthUtils.isCurrentUserAuthenticated()) {
+            Log.w(TAG, "user is not authenticated! This should not happen!");
+            return;
+        }
+
+        bookPhotosListAdapter = new BookPhotosListAdapter();
+        bookPhotosGalleryRecyclerView = findViewById(R.id.book_photos_gallery);
+        GridLayoutManager layoutManager = new GridLayoutManager(getApplicationContext(), 3);
+        ItemOffsetDecoration itemDecoration = new ItemOffsetDecoration(getApplicationContext(), R.dimen.item_offset);
+        bookPhotosGalleryRecyclerView.addItemDecoration(itemDecoration);
+        bookPhotosGalleryRecyclerView.setAdapter(bookPhotosListAdapter);
+        bookPhotosGalleryRecyclerView.setLayoutManager(layoutManager);
+
+        Log.i(TAG, "bookID: " + bookID);
+        BookPhotosViewModelFactory factory = new BookPhotosViewModelFactory(bookID);
+        bookPhotosViewModel = ViewModelProviders.of(this, factory).get(BookPhotosViewModel.class);
+        bookPhotosViewModel.getPhotosLiveData().observe(this, new Observer<ArrayList<Photo>>() {
+            @Override
+            public void onChanged(@Nullable ArrayList<Photo> photos) {
+                bookPhotosListAdapter.submitList(photos);
+            }
+        });
+
+        if (FirebaseAuthUtils.getCurrentUser().getUid().equals(ownerUserID)) {
+            cameraHandler = CameraHandler.newInstance(this);
+
+            bookPhotosListAdapter.setOnItemClickListener(new BookPhotosListAdapter.OnItemClickListener() {
+                @Override
+                public void onClick(final View v, final Photo photo) {
+                    cameraHandler.dispatchTakePictureIntent();
+                    cameraHandler.setPictureTakenListener(new CameraHandler.OnPictureTakenListener() {
+                        @Override
+                        public void onPictureTaken(Bitmap bitmap) {
+                            bookPhotosViewModel.updatePhoto(photo, bitmap);
+                            cameraHandler.removePictureTakenListener();
+                        }
+                    });
+                }
+
+                @Override
+                public void onLongClick(View v, final Photo photo) {
+                    // Code gotten from https://stackoverflow.com/a/33650105
+                    // License: https://creativecommons.org/licenses/by-sa/3.0/
+                    AlertDialog.Builder builder = new AlertDialog.Builder(v.getContext());
+                    builder.setMessage(getString(R.string.book_photo_dialog_prompt))
+                            .setPositiveButton(getString(R.string.dialog_delete), new DialogInterface.OnClickListener() {
+                                @Override
+                                public void onClick(DialogInterface dialog, int which) {
+                                    bookPhotosViewModel.deletePhoto(photo);

```

Comment:
Photo is not removed when returning back to the edit activity. It seems it's not being updated. 
It does change once you exit and re-enter the edit page..

Reply:
I think it's because in the AddEdiBooktActivity, I do a null check when I observe for changes to the first book photo. I don't update the image view when it's null but null is returned when the data doesn't exist. That should be an easy fix.

Reply:
Fixed using Glide. Glide is really cool, if you give it a fallback picture and the bitmap you try to load is null, it just loads the fallback picture.